openedx/core/djangoapps/import_export/courses/tests/utils.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=E1101
+'''
+Utilities for contentstore tests
+'''
+
+from datetime import timedelta
+
+from django.utils import timezone
+from rest_framework.test import APIClient
+
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from django.conf import settings
+
+from provider.oauth2.models import AccessToken, Client as OAuth2Client
+from provider import constants
+
+TEST_DATA_DIR = settings.COMMON_TEST_DATA_ROOT
+
+
+def create_oauth2_client(user):
+    """
+    Create an OAuth2 client associated with the given user and generate an
+    access token for said client.
+
+    :param user:
+    :return: a Client (provider.oauth2) and an AccessToken
+    """
+    # Register an OAuth2 Client
+    client = OAuth2Client(
+        user=user,
+        name=user.username,
+        url="http://127.0.0.1/",
+        redirect_uri="http://127.0.0.1/",
+        client_type=constants.CONFIDENTIAL
+    )
+    client.save()
+
+    # Generate an access token for the client
+    access_token = AccessToken(
+        user=user,
+        client=client,
+
+        # Set the access token to expire one day from now
+        expires=timezone.now() + timedelta(1, 0),
+        scope=constants.READ_WRITE
+    )
+    access_token.save()
+
+    return client, access_token
+
+
+def use_access_token(client, access_token):
+    """
+    Make an APIClient pass an access token for all requests
+
+    :param client: an APIClient
+    :param access_token: an AccessToken
+    """
+    client.credentials(
+        HTTP_AUTHORIZATION="Bearer {}".format(access_token.token)
+    )
+
+    return client
+
+
+class CourseTestCase(ModuleStoreTestCase):
+    """
+    Extendable base for test cases dealing with courses
+    """
+    def setUp(self):
+        """
+        These tests need a user in the DB so that the django Test Client can
+        log them in.
+        The test user is created in the ModuleStoreTestCase setUp method.
+        They inherit from the ModuleStoreTestCase class so that the mongodb
+        collection will be cleared out before each test case execution and
+        deleted afterwards.
+        """
+        self.user_password = super(CourseTestCase, self).setUp()
+
+        # Create an APIClient to simulate requests (like the Django Client, but
+        # without CSRF)
+        api_client = APIClient()
+
+        # Register an OAuth2 Client
+        # pylint: disable=unused-variable
```

Comment:
You can prefix variable names with an underscore to indicate you know they are unused. Please do that rather than disable this warning.

If you absolutely _must_ use disables, please do them inline so that the warning is only suppressed for that one line. With this fashion, you are disabling for the remainder of the function.